From 5bdaf17b05b7cb9620681cbd594ec050237e2403 Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Tue, 18 Jan 2022 23:57:24 +0000
Subject: [PATCH] makefile: avoid ldconfig when cross-compiling

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

Also create necessary symlinks for `libnvidia-container-go.so` if the
package is compiled `WITH_NVCGO=yes`.

Signed-off-by: Markus Boehme <markubo@amazon.com>
---
 Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 60429d3..7d4c19b 100644
--- a/Makefile
+++ b/Makefile
@@ -269,7 +269,10 @@ ifeq ($(WITH_NVCGO), yes)
 	$(INSTALL) -m 755 $(DEPS_DIR)$(libdir)/$(LIBGO_SHARED) $(DESTDIR)$(libdir)
 	$(LN) -sf $(LIBGO_SONAME) $(DESTDIR)$(libdir)/$(LIBGO_SYMLINK)
 endif
-	$(LDCONFIG) -n $(DESTDIR)$(libdir)
+	$(LN) -sf $(LIB_SHARED) $(DESTDIR)$(libdir)/$(LIB_SONAME)
+ifeq ($(WITH_NVCGO), yes)
+	$(LN) -sf $(LIBGO_SHARED) $(DESTDIR)$(libdir)/$(LIBGO_SONAME)
+endif
 	# Install configuration files
 	$(MAKE_DIR)/$(LIB_PKGCFG).in "$(strip $(VERSION))" "$(strip $(LIB_LDLIBS_SHARED))" > $(DESTDIR)$(pkgconfdir)/$(LIB_PKGCFG)
 	# Install binary files
-- 
2.39.0

